packages/cozy-harvest-lib/src/models/ConnectionFlow.spec.js

```diff
@@ -388,6 +389,34 @@ describe('ConnectionFlow', () => {
       expect(flow.account).toEqual(fixtures.updatedAccount)
     })
 
+    it('should add the defaultFolderPath to the account', async () => {
+      const { flow, client } = setup()
+      prepareTriggerAccount.mockResolvedValue(fixtures.updatedAccount)
+
+      ensureTrigger.mockImplementation(
+        async () => fixtures.createdTriggerWithFolder.attributes
+      )
+      client.query.mockImplementation(async () => ({
+        data: {
+          path: '/default/folder/path'
+        }
+      }))
+      await flow.ensureTriggerAndLaunch(client, {
+        account: fixtures.existingAccount,
+        trigger: fixtures.existingTrigger,
+        konnector: fixtures.konnectorWithFolder
+      })
+      expect(flow.account).toEqual(fixtures.updatedAccount)
+      expect(saveAccount).toHaveBeenCalledWith(
+        client,
+        fixtures.konnectorWithFolder,
+        {
+          ...fixtures.existingAccount,
+          defaultFolderPath: '/default/folder/path'
+        }
+      )
+    })
```

Comment:
let's test also those cases:
- folderId not defined
- account.defaultFolderPath exists
- what if client.query return empty/undefined (I don't know if it's possible)

Reply:
- folderId not defined : already tested implicitely in the other tests of the suite
- the stack will ensure itself the destination folder exists and use defaultFolderPath to name it. This is not the responsibility of harvest to check this
- this is not supposed to happend since it is the responsibility of ensureTrigger, before, to also ensure the destination folder exists. But you are right, i will at least catch the exception in this case with a warning log.